sormas-ui/src/main/java/de/symeda/sormas/ui/task/TaskController.java

```diff
@@ -177,4 +184,137 @@ public void run() {
 				});
 		}
 	}
+
+	private DistrictReferenceDto getTaskIndexDistrict(TaskIndexDto taskDto) {
+		DistrictReferenceDto district = null;
+		if (taskDto.getCaze() != null) {
+			CaseDataDto caseDto = FacadeProvider.getCaseFacade().getCaseDataByUuid(taskDto.getCaze().getUuid());
+			district = caseDto.getDistrict();
+		} else if (taskDto.getContact() != null) {
+			ContactDto contactDto = FacadeProvider.getContactFacade().getContactByUuid(taskDto.getContact().getUuid());
+			if (contactDto.getRegion() != null && contactDto.getDistrict() != null) {
+				district = contactDto.getDistrict();
+			} else {
+				CaseDataDto caseDto = FacadeProvider.getCaseFacade().getCaseDataByUuid(contactDto.getCaze().getUuid());
+				district = caseDto.getDistrict();
+			}
+		} else if (taskDto.getEvent() != null) {
+			EventDto eventDto = FacadeProvider.getEventFacade().getEventByUuid(taskDto.getEvent().getUuid());
+			district = eventDto.getEventLocation().getDistrict();
+		}
+		return district;
+	}
+
+	public void showBulkTaskDataEditComponent(Collection<? extends TaskIndexDto> selectedTasks) {
+		if (selectedTasks.size() == 0) {
+			new Notification(
+				I18nProperties.getString(Strings.headingNoTasksSelected),
+				I18nProperties.getString(Strings.messageNoTasksSelected),
+				Type.WARNING_MESSAGE,
+				false).show(Page.getCurrent());
+			return;
+		}
+
+		// Check if tasks with multiple districts have been selected
+		DistrictReferenceDto district = null;
+		for (TaskIndexDto taskIndexDto : selectedTasks) {
```

Comment:
Just realised that this will perform poorly and put a lot of stress onto the database when many tasks are selected since it requires potentially one additional query per selected task; please re-write `getTaskIndexDistrict` to instead return whether or not all tasks lie in the same OR no district for a list of given tasks (instead of one single task); that code should then also be moved to the backend